Add TE fuser ops

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68826
* __->__ #68825

Factoring out the elementwise ops in tensorexpr fuser and adding their corresponding shape functions, since we need shape functions to fuse them with dynamic shapes

Differential Revision: [D32732466](https://our.internmc.facebook.com/intern/diff/D32732466)

Commit:
Add TE fuser ops

[ghstack-poisoned]

ZolotukhinM commented:
Do you mind adding more details to the commit message?

Commit:
Update on "Add TE fuser ops"


Factoring out the elementwise ops in tensorexpr fuser and adding their corresponding shape functions, since we need shape functions to fuse them with dynamic shapes


[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/runtime/symbolic_shape_registry_util.cpp:
```diff
@@ -0,0 +1,124 @@
+
+
```
delet dis

ZolotukhinM commented on torch/csrc/jit/runtime/symbolic_shape_registry_util.cpp:
```diff
@@ -0,0 +1,124 @@
+const OperatorMap<std::string>& get_tensorexpr_elementwise_set() {
+  // clang-format off
+ static const OperatorMap<std::string> tensorexpr_elementwise_set{
+     {"aten::add.Scalar(Tensor self, Scalar other, Scalar alpha=1) -> Tensor", "unary"},
+      {"aten::_cast_Float(Tensor self, bool non_blocking) -> Tensor", "unary"},
```
Nit: formatting seems 1 char off here.

ZolotukhinM commented on torch/csrc/jit/runtime/symbolic_shape_registry_util.cpp:
```diff
@@ -0,0 +1,124 @@
+      {"aten::lerp.Tensor(Tensor self, Tensor end, Tensor weight) -> Tensor", "broadcast_three"},
+      {"aten::addcmul(Tensor self, Tensor tensor1, Tensor tensor2, *, Scalar value=1) -> Tensor", "broadcast_three"},
+      {"aten::where.self(Tensor condition, Tensor self, Tensor other) -> Tensor", "broadcast_three"},
+      {"aten::where.ScalarSelf(Tensor condition, Scalar self, Tensor other) -> Tensor", "broadcast_one_three"},
+      // TODO: enable slice, shape inference is not implemented for this op yet
```
The list didn't change during the move, right?

eellison replied:
yes - just copy-pasting

eellison replied:
no, ctrl-F "aten::" 98 results in both lists

ZolotukhinM reviewed (approved):
Looks good!

Commit:
Update on "Add TE fuser ops"


Factoring out the elementwise ops in tensorexpr fuser and adding their corresponding shape functions, since we need shape functions to fuse them with dynamic shapes


[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32732466).
